### PR TITLE
week1 databases assignement  complete

### PR DESCRIPTION
**Pull request by Hailemariam7**

### CODE REVIEW

File: Week1/Assignment/world.js [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+  }
+);
+
+connection.query("SELECT Population FROM country", (err, results) => {
+  if (err) throw err;
+  console.log("10. Population of the world:");
+  let totalPopulation = 0;
+  results.forEach((country) => (totalPopulation += country.Population));
+  console.log(totalPopulation);
+});
```
Comment (codethenpizza) [inline Week1/Assignment/world.js:105]: You get the right result, but using code. 
How would you do this only with query?

File: Week1/Assignment/studentMeeting.sql [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+);
+
+CREATE TABLE IF NOT EXISTS Meeting (
+  meeting_no INT AUTO_INCREMENT PRIMARY KEY,
+  meeting_title VARCHAR(255),
+  starting_time DATETIME,
+  ending_time DATETIME,
+  room_no INT,
+  FOREIGN KEY (room_no) REFERENCES Room(room_no)
+);
```
Comment (codethenpizza) [inline Week1/Assignment/studentMeeting.sql:24]: One thing I noticed is that some of your columns don't have the "NOT NULL" constraint. Adding this constraint can help ensure data integrity by making sure that important fields always have a value.
 
Here is a suggestion on where you might want to add "NOT NULL":
```
invitee_name VARCHAR(255) NOT NULL  -- Names are usually required
```
What do you think, what are the other fields that sounds like suppose to be required?

**Review by codethenpizza — CHANGES_REQUESTED** [review]
You're doing great. There are a few things to consider that can make your code even better:

- DRY Principle: Avoid repeating the same pattern multiple times by creating a helper function for your queries. This makes your code cleaner and easier to maintain.

- Data Integrity: Add "NOT NULL" constraints to columns where it makes sense to ensure that important fields always have values. This can help maintain data integrity and prevent potential issues.


Also, as I see there's no direct requirement to create tables and data via JS, but if you feel like you want a bit more challenge, you can execute your queries with JS code 🙂

File: Week1/Assignment/studentMeeting.sql (hunk 2 above)
Comment (Hailemariam7) [reply Week1/Assignment/studentMeeting.sql:24]: I have corrected it. Now meeting title, start_time and end_time among other columns have NOT NULL added.

File: Week1/Assignment/world.js (hunk 1 above)
Comment (Hailemariam7) [reply Week1/Assignment/world.js:105]: certainly, it is shorter with query. I have changed it.  thanks!

File: Week1/Assignment/world.js [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+);
+
+connection.query(
+  'SELECT Name FROM country WHERE Continent = "Europe"',
+  (err, results) => {
+    if (err) throw err;
+    console.log("4. Countries in Europe:");
+    console.log(results);
+  }
+);
```
[the remark this answers is not in the document]
Comment (Hailemariam7) [reply Week1/Assignment/world.js:52]: I have tried to shorten it now by implement the connection.query in a separate function, I hope it is more readable now and cleaner.

**Comment by Hailemariam7** [comment]
Thank you very much for taking you valuable time to review this! I have implemented the feedbacks. I hope it good enough to be approved, but open for more feedback at the same time. I will make the PR now.

**Review by codethenpizza — APPROVED** [review]

**Comment by codethenpizza** [comment]
Good job!

**Comment by Hailemariam7** [comment]
Thank you

**Pull request closed without merging**